code/modules/power/catalytic_engine.dm

```diff
@@ -0,0 +1,521 @@
+#define GEN_ANODE 1
+#define GEN_CATHODE 2
+
+#define UNIT_OPEN 0
+#define UNIT_INACTIVE 1
+#define UNIT_ACTIVE 2
+
+/obj/decal/fakeobjects/catalytic_doodad
+	name = "catalytic generator pipe"
+	desc = "Pipe section allowing a catalytic rod to contact outside fluid for catalysis."
+	icon = 'icons/obj/machines/catalysis.dmi'
+	icon_state = "doodad"
+	anchored = 1
+
+/obj/machinery/power/catalytic_generator
+	name = "catalytic generator core"
+	desc = "Harnesses catalysts' reactions with a large body of appropriate fluid to generate electricity."
+	icon = 'icons/obj/machines/catalysis.dmi'
+	icon_state = "core"
+	anchored = 1
+	density = 1
+
+	var/obj/machinery/catalytic_rod_unit/anode_unit
+	var/obj/machinery/catalytic_rod_unit/cathode_unit
+
+	var/gen_rate = 0
+	var/defer_updateicon = FALSE
+	var/output_tier = 0
+	var/running = 0
```

Comment:
0 or false

Reply:
Running should be true/false, yep. Remainder are numbers (which are 0 while inactive).